ambry-router/src/main/java/com.github.ambry.router/DeleteOperation.java

```diff
@@ -163,9 +167,14 @@ void handleResponse(ResponseInfo responseInfo, DeleteResponse deleteResponse) {
     routerMetrics.getDataNodeBasedMetrics(replica.getDataNodeId()).deleteRequestLatencyMs.update(requestLatencyMs);
     // Check the error code from NetworkClient.
     if (responseInfo.getError() != null) {
```

Comment:
I tried. but the issue is that, sometimes we set router error state depending on the situtation. For eg, when there is not response at all, we set some error. When correlationId mismatches, we set some other router exception. So, I went with this approach. I tried logging within processServerError(), but then have to pass in all other info like replicaId and correlationId. so, felt this is the best way.